src/modules/core/components/dialog-full-screen/dialog-full-screen.tsx

```diff
@@ -0,0 +1,330 @@
+        visibilityPercent = 100;
+      }
+
+      if (visibilityPercent < 0) {
+        visibilityPercent = 0;
+      }
+
+      const contentCss =
+        config.content.getCssByVisibilityPercent(visibilityPercent);
+      const overlayCss =
+        config.overlay.getCssByVisibilityPercent(visibilityPercent);
+
+      reassignCssProperties(contentRef.current, contentCss);
+      reassignCssProperties(overlayRef.current, overlayCss);
+    },
+    [config.content, config.overlay]
+  );
+
+  const handleContentTouchStart = useCallback(
+    (e: React.TouchEvent<HTMLDivElement>) => {
+      if (!contentRef.current || !overlayRef.current) {
+        return;
+      }
+
+      // @ts-ignore
+      const classList: string[] = (e.target && [...e.target.classList]) || [];
+
+      const isIgnore = classList.some((className) =>
+        SWIPES_IGNORE_CLASS_NAMES.includes(className)
+      );
+
+      if (isIgnore) {
+        return;
+      }
+
+      dragContentRect.current = contentRef.current.getBoundingClientRect();
+      pointerXPosition.current = e.touches[0].clientX;
+      pointerYPosition.current = e.touches[0].clientY;
+
+      contentRef.current.style.transition = 'none';
+      overlayRef.current.style.transition = 'none';
+
+      document.addEventListener('touchmove', handleContentDrag);
+    },
+    [contentRef, handleContentDrag]
+  );
+
+  const handleContentTouchEnd = useCallback(() => {
+    if (!contentRef.current || !overlayRef.current) {
+      return;
+    }
+
+    const x = contentRef.current.getBoundingClientRect().x;
+
+    const CLOSE_OFFSET_BREAKPOINT = 60;
+
+    if (x - pointerXPosition.current < CLOSE_OFFSET_BREAKPOINT) {
+      // back position
+      contentRef.current.style.transition = 'all 0.3s';
+      overlayRef.current.style.transition = 'all 0.3s';
+
+      reassignCssProperties(contentRef.current, config.content.cssOpen);
+      reassignCssProperties(overlayRef.current, config.overlay.cssOpen);
+    } else {
+      // hide
+      contentAnimationController.set(
+        getElementCssProperties(contentRef.current, config.content.cssHidden)
+      );
+      overlayAnimationController.set(
+        getElementCssProperties(overlayRef.current, config.overlay.cssHidden)
+      );
+      Promise.all([
+        contentAnimationController.start({
+          ...config.content.cssHidden,
+        }),
+        overlayAnimationController.start({
+          ...config.overlay.cssHidden,
+        }),
+      ]).finally(() => {
+        _isOpen.setFalse();
+        onOpenChange && onOpenChange(false);
+      });
+    }
+
+    dragContentRect.current = null;
+    pointerXPosition.current = 0;
+    pointerYPosition.current = 0;
+    document.removeEventListener('touchmove', handleContentDrag);
+  }, [
+    _isOpen.setFalse,
+    config.content.cssHidden,
+    config.content.cssOpen,
+    config.overlay.cssHidden,
+    contentAnimationController,
+    handleContentDrag,
+    onOpenChange,
+    overlayAnimationController,
+  ]);
```

Comment:
It would be nice to refactor it somehow, maybe leave a todo for the future or add some comment?

Reply:
left comment